### DISCUSSION TITLE
How to use a provider inside a custom hook?

### DISCUSSION THREAD

**Discussion by yom-ko** [rrousselGit/riverpod#913] [Q&A] [answered]
In ` hooks_riverpod ^0.14.0+5`, if I needed to access a provider inside a custom hook it was straightforward, like:

```dart
useOpenAd() {
...
final backToHomeCount = useProvider(
      settingsProvider.select((settings) => settings.backToHomeCount));
...
}
```

But with the latest version `1.0.0`, I can't see a simple solution. According to the docs, the `ref` object can only be obtained in another provider, in all kinds of widgets, or in the `HookConsumer`. Am I missing something? Is it possible to obtain `ref` (or otherwise access a provider ) inside a [custom hook](https://pub.dev/packages/flutter_hooks#how-to-use)?
TIA!

**Comment by rrousselGit** [comment, marked as answer]
You could pass a WidgetRef obtained from a ConsumerWidget/Consumer as a parameter to your custom hook, or straight up pass the resolved value.

**Reply by yom-ko** [reply]
Thanks, I thought about passing the ref as a hook's parameter but then I saw [this warning](https://riverpod.dev/docs/concepts/reading#obtaining-a-ref-from-a-widget) in the docs:

> When a widget obtains a "ref", this "ref" should not be passed around. It should be used only by the widget that created this object.

In this case, will the returned value be updated normally whenever the provider state changes? No gotchas here?

**Reply by rrousselGit** [reply]
Yes, that should be fine.
Passing the ref to hooks is kind of an exception to this warning
